third_party/java/arcs/build_defs/internal/kotlin.bzl

```diff
@@ -97,6 +98,16 @@ DEFAULT_LIBRARY_PLATFORMS = ["jvm"]
 # Default set of platforms for Kotlin particles.
 DEFAULT_PARTICLE_PLATFORMS = ["jvm"]
 
+# Controls output of type slicing interfaces for entities in generated particle classes.
+# TODO(b/182330900): hard-code to 'on' once type slicing has fully launched.
+def set_type_slicing_flag():
```

Comment:
This method name sounds like a setter, but it's more like a getter. How about something like `is_type_slicing_enabled()` instead?

I'd also suggest adding a comment explaining that this flag uses the "default" value of the flag (NOT_READY vs READY) to control the logic, rather than being overridden by clients. (That should probably go in its definition in flags.bzl too.)

Reply:
Actually, now that I think about it some more, this should probably check for LAUNCHED instead of READY. That is the status we use for features that are fully launched and are no longer overrideable by clients.

Reply:
Done.